wasm/jit/jit_arm64_test.go

```diff
@@ -351,6 +353,193 @@ func TestArm64Compiler_loadValueOnStackToRegister(t *testing.T) {
+						{math.Float64bits(math.Inf(-1)), math.Float64bits(math.NaN())},
+						{math.Float64bits(math.NaN()), math.Float64bits(math.Inf(1))},
+						{math.Float64bits(math.NaN()), math.Float64bits(math.Inf(-1))},
+					} {
+						x1, x2 := values[0], values[1]
+						t.Run(fmt.Sprintf("x1=0x%x,x2=0x%x", x1, x2), func(t *testing.T) {
+							env := newJITEnvironment()
+							compiler := env.requireNewCompiler(t)
+							err := compiler.emitPreamble()
+							require.NoError(t, err)
+
+							// Emit consts operands.
+							for _, v := range []uint64{x1, x2} {
+								switch signedType {
+								case wazeroir.SignedTypeUint32:
+									err = compiler.compileConstI32(&wazeroir.OperationConstI32{Value: uint32(v)})
+								case wazeroir.SignedTypeInt32:
+									err = compiler.compileConstI32(&wazeroir.OperationConstI32{Value: uint32(int32(v))})
+								case wazeroir.SignedTypeInt64, wazeroir.SignedTypeUint64:
+									err = compiler.compileConstI64(&wazeroir.OperationConstI64{Value: v})
+								case wazeroir.SignedTypeFloat32:
+									err = compiler.compileConstF32(&wazeroir.OperationConstF32{Value: math.Float32frombits(uint32(v))})
+								case wazeroir.SignedTypeFloat64:
+									err = compiler.compileConstF64(&wazeroir.OperationConstF64{Value: math.Float64frombits(v)})
+								}
+								require.NoError(t, err)
+							}
+
+							// At this point, two values exist for comparison.
+							require.Equal(t, uint64(2), compiler.locationStack.sp)
+
+							// Emit the operation.
+							switch kind {
+							case wazeroir.OperationKindLe:
+								err = compiler.compileLe(&wazeroir.OperationLe{Type: signedType})
+							case wazeroir.OperationKindLt:
+								err = compiler.compileLt(&wazeroir.OperationLt{Type: signedType})
+							case wazeroir.OperationKindGe:
+								err = compiler.compileGe(&wazeroir.OperationGe{Type: signedType})
+							case wazeroir.OperationKindGt:
+								err = compiler.compileGt(&wazeroir.OperationGt{Type: signedType})
+							}
+							require.NoError(t, err)
+
+							// We consumed two values, but push the result back.
+							require.Equal(t, uint64(1), compiler.locationStack.sp)
+							resultLocation := compiler.locationStack.peek()
+							// Plus the result must be located on a conditional register.
+							require.True(t, resultLocation.onConditionalRegister())
+
+							// Move the conditional register value to a general purpose register to verify the value.
+							compiler.loadConditionalRegisterToGeneralPurposeRegister(resultLocation)
+							require.True(t, resultLocation.onRegister())
+
+							// Release the value to the memory stack again to verify the operation, and then return.
+							compiler.releaseRegisterToStack(resultLocation)
+							compiler.returnFunction()
+
+							// Generate the code under test.
+							code, _, _, err := compiler.compile()
+							require.NoError(t, err)
+
+							// Run code.
+							env.exec(code)
+
+							// Check the stack.
```

Comment:
be more specific, as the assertion on stackPointer says we are checking it somehow

```suggestion
							// There should only be one value on the stack
```